### ISSUE TITLE
Unhandled ObjectDisposedException in SuperSocket.ProcessConnect on Mono 4.8.0

### ISSUE THREAD

**Issue by jlewin** [kerryjiang/WebSocket4Net#86]
We're running into unhandled exceptions that we can't guard against on Mono 4.8.0 clients as they process through their Succeeded/ProcessConnect logic. See https://github.com/MatterHackers/MatterControl/issues/1989 for reference. 

Not sure what to try but it repro's consistently after connecting to our SocketIO server and only on Mono 4.8.0. In the past 6 months we've locked out Mono clients down to 4.4.2 because Mono 4.6.x would fail to connect. Now that 4.8.0 is out it just crashes.

**System.ObjectDisposedException: Cannot access a disposed object.**
```
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.Net.Sockets.Socket'.
  at System.Net.Sockets.Socket.ThrowIfDisposedAndClosed () [0x00021] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.Socket.GetSocketOption (System.Net.Sockets.SocketOptionLevel optionLevel, System.Net.Sockets.SocketOptionName optionName) [0x00000] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at SuperSocket.ClientEngine.TcpClientSession.ProcessConnect (System.Net.Sockets.Socket socket, System.Object state, System.Net.Sockets.SocketAsyncEventArgs e) [0x00060] in <4ba236fbf29248a4985a92e6e3907e4a>:0 
  at SuperSocket.ClientEngine.AsyncTcpSession.SocketEventArgsCompleted (System.Object sender, System.Net.Sockets.SocketAsyncEventArgs e) [0x0000c] in <4ba236fbf29248a4985a92e6e3907e4a>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.OnCompleted (System.Net.Sockets.SocketAsyncEventArgs e) [0x00014] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.FinishWrapperConnectSuccess (System.Net.Sockets.Socket connectSocket, System.Int32 bytesTransferred, System.Net.Sockets.SocketFlags flags) [0x00016] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.MultipleConnectAsync.Succeed () [0x0002d] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.MultipleConnectAsync.InternalConnectCallback (System.Object sender, System.Net.Sockets.SocketAsyncEventArgs args) [0x000d4] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.OnCompleted (System.Net.Sockets.SocketAsyncEventArgs e) [0x00014] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.Complete () [0x00000] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.Socket.<ConnectAsyncCallback>m__3 (System.IAsyncResult ares) [0x00062] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.SocketAsyncResult+<Complete>c__AnonStorey0.<>m__0 (System.Object _) [0x00000] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00019] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00096] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.Net.Sockets.Socket'.
  at System.Net.Sockets.Socket.ThrowIfDisposedAndClosed () [0x00021] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.Socket.GetSocketOption (System.Net.Sockets.SocketOptionLevel optionLevel, System.Net.Sockets.SocketOptionName optionName) [0x00000] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at SuperSocket.ClientEngine.TcpClientSession.ProcessConnect (System.Net.Sockets.Socket socket, System.Object state, System.Net.Sockets.SocketAsyncEventArgs e) [0x00060] in <4ba236fbf29248a4985a92e6e3907e4a>:0 
  at SuperSocket.ClientEngine.AsyncTcpSession.SocketEventArgsCompleted (System.Object sender, System.Net.Sockets.SocketAsyncEventArgs e) [0x0000c] in <4ba236fbf29248a4985a92e6e3907e4a>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.OnCompleted (System.Net.Sockets.SocketAsyncEventArgs e) [0x00014] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.FinishWrapperConnectSuccess (System.Net.Sockets.Socket connectSocket, System.Int32 bytesTransferred, System.Net.Sockets.SocketFlags flags) [0x00016] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.MultipleConnectAsync.Succeed () [0x0002d] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.MultipleConnectAsync.InternalConnectCallback (System.Object sender, System.Net.Sockets.SocketAsyncEventArgs args) [0x000d4] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.OnCompleted (System.Net.Sockets.SocketAsyncEventArgs e) [0x00014] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.Complete () [0x00000] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.Socket.<ConnectAsyncCallback>m__3 (System.IAsyncResult ares) [0x00062] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Net.Sockets.SocketAsyncResult+<Complete>c__AnonStorey0.<>m__0 (System.Object _) [0x00000] in <5641e4edad4f4464ba58c620a7b8ea48>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00019] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00096] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
```

**Comment by AlexanderSheety** [comment]
Hello Kerry!
I have pcl xamarin project!
When i build uwp project will run with this errors: 

> 1.Some packages are not compatible with UAP,Version=v10.0 (win10-x86).			
> 2.System.Net.Security 4.0.0 provides a compile-time reference assembly for System.Net.Security on UAP,Version=v10.0, but there is no run-time assembly compatible with win10-x86.		

My Pcl project.json:

```
{
  "supports": {},
  "dependencies": {
    "NETStandard.Library": "1.6.1",
    "WebSocket4Net": "0.15.0-beta6",
    "Xamarin.Forms": "2.3.3.193"
  },
  "frameworks": {
    "netstandard1.4": {
      "imports": "portable-net45+win8+wpa81+wp8"
    }
  }
}
```

My UWP project.json:
```
{
  "dependencies": {
    "Microsoft.NETCore.UniversalWindowsPlatform": "5.2.2"
  },
  "frameworks": {
    "uap10.0": {"imports": [ "netstandard1.4"]}
  },
  "runtimes": {
    "win10-x86": {},
  }
}
```

**Comment by kerryjiang** [comment]
OK, I'll check this issue!

**Comment by kerryjiang** [comment]
Did a fix for this issue.
https://github.com/kerryjiang/SuperSocket.ClientEngine/commit/d16adc1779c079589346363f9ce596ca973e706c

I'll release a new version of SuperSocket.ClientEngine and WebSocket4Net.

**Comment by kerryjiang** [comment]
I released a new version in nuget including this fix 0.15.0-beta8, please have a try.

**Comment by kerryjiang** [comment]
It should has been fixed in the latest release 0.15;
https://www.nuget.org/packages/WebSocket4Net/0.15.0
